rustls-cert-gen/src/cert/mod.rs

```diff
@@ -0,0 +1,44 @@
+use rcgen::{Certificate, CertificateParams, Error};
+use std::{fs::File, io, path::Path};
+
+mod params;
+pub use params::BuildParams;
+
+mod ca;
+mod entity;
+mod signature;
+
+pub trait CertParams {
+	fn params(&self) -> &CertificateParams;
+}
+
+#[derive(Debug, Clone)]
+/// Struct to represent a Serialized Cert/Key pair
+pub struct SerializedEntity {
+	pub cert_pem: String,
+	pub key_pem: String,
+}
+
+impl SerializedEntity {
+	pub fn write(&self, dir: &Path, name: &str) -> Result<(), io::Error> {
```

Comment:
I think this will fail if `dir` doesn't exist. Maybe add a `std::fs::create_dir_all(dir)?;`